satpy/readers/eum_l2_bufr.py

```diff
@@ -84,54 +94,95 @@ class SeviriL2BufrFileHandler(BaseFileHandler):
+            timeStr = attr['typicalDate']+attr['typicalTime']
+
+            darr = self.get_array('satelliteIdentifier')
+            arr = da.where(darr != BUFR_FILL_VALUE, darr, darr).compute()
+            if (arr.ndim > 0):
+                sc_id = int(arr[0])
+            else:
+                sc_id = int(arr)
+
+            self.bufr_header = {}
+            self.bufr_header['NominalTime'] = datetime.strptime(timeStr, "%Y%m%d%H%M%S")
+            self.bufr_header['SpacecraftName'] = data_center_dict[sc_id]['name']
+            self.bufr_header['RectificationLongitude'] = data_center_dict[sc_id]['ssp']
 
         if rectification_longitude != 'default':
-            self.mpef_header['RectificationLongitude'] = f'E{int(rectification_longitude * 10):04d}'
-
-        self.with_adef = with_area_definition
-        if self.with_adef and filetype_info['file_type'] == 'seviri_l2_bufr_amv':
-            logging.warning("AMV BUFR data cannot be loaded with an area definition. Setting self.with_def = False.")
+            self.bufr_header['RectificationLongitude'] = f'E{int(rectification_longitude * 10):04d}'
+
+        self.filetype = filetype_info['file_type']
+        self.seg_size = seg_size_dict[self.filetype]
+        if self.seg_size:
+            # make this keyword not usable for non-grided products
+            self.with_adef = with_area_definition
+
+            darr = self.get_array('segmentSizeAtNadirInXDirection')
+            # This key may not exists in some files
+            # This is specifically a "dirty" fox for the unit tests because the template used doesn't
+            # contain that key so initializing the object would fail without thic check
+            # but 'real' EUM BUFR files shall all have it
+            if darr is None:
+                self.resolution = None
+            else:
+                arr = da.where(darr != BUFR_FILL_VALUE, darr, darr).compute()
+                self.resolution = arr[0]
+
+            # Note: There a difference between resolution and seg_size as from the seg_size_dict
+            # in the bufr files segmentSizeAtNadirInXDirection is encoded in meters so it's indeed
+            # the physical size of the segment while seg_size is the number of pixel lines / cols
+            # but because FCI pixel size is 1km they look very similar, modulo 1000
+        else:
             self.with_adef = False
-
-        self.seg_size = seg_size_dict[filetype_info['file_type']]
+            self.resolution = None
 
     @property
     def start_time(self):
         """Return the repeat cycle start time."""
-        return self.mpef_header['NominalTime']
+        return self.bufr_header['NominalTime']
 
-    @property
-    def end_time(self):
-        """Return the repeat cycle end time."""
-        return self.start_time + timedelta(minutes=15)
+    # @property
+    # def end_time(self):
+    #     """Return the repeat cycle end time."""
+    #     return self.start_time + timedelta(minutes=15)
```

Comment:
I think it's fine to delete if for know, as it may soon get outdated if the satellite change their location and scanning frequency.  We can re-introduce it later if needed.